org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/ClasspathJsr199.java

```diff
@@ -156,12 +160,29 @@ public char[][][] findTypeNames(String aQualifiedPackageName, String moduleName)
 	public void initialize() throws IOException {
 		if (this.jrt != null) {
 			this.jrt.initialize();
+		} else if (this.location.isModuleOrientedLocation()) {
+			Iterable<Set<Location>> locationsForModules = this.fileManager.listLocationsForModules(this.location);
+			for (Set<Location> locs: locationsForModules) {
+				for (Location loc : locs) {
+					if (loc instanceof LocationWrapper wrapper) {
```

Comment:
This use of LocationWrapper leaks an abstraction, it will not work with other file managers.

But then, how to find the contained paths??

Reply:
If it were a StandardJavaFileManager we could use `getLocationAsPaths(Location)`

Reply:
I now have alternative implementations using `this.fileManager.list()`, i.e., relying neither on `LocationWrapper` nor on `StandardJavaFileManager`.